crates/sui-core/src/consensus_adapter.rs

```diff
@@ -231,13 +282,29 @@ impl ConsensusListener {
                     let outcome = result.map_err(SuiError::from);
                     let digest = Self::hash_serialized_transaction(&serialized);
                     if let Some(repliers) = self.pending.remove(&digest) {
-                        for replier in repliers {
+                        for (_, replier) in repliers {
                             if replier.send(outcome.clone()).is_err() {
                                 debug!("No replier to listen to consensus output {digest}");
                             }
                         }
                     }
                 }
+
+                Some((digest, id)) = closed_notifications.next() => {
+                    let should_delete = if let Some(list) = self.pending.get_mut(&digest) {
+                        // First clean up the list
+                        list.retain(|(item_id, _)| *item_id != id);
```

Comment:
Should we use an HashMap rather than a Vec?

Reply:
This is usually 1-2 elements, so I do not think that a HashMap is really needed here.